### DISCUSSION TITLE
请问，现在接入 next-auth 之后跟登陆之前有什么区别？

### DISCUSSION THREAD

**Discussion by DrayChou** [lobehub/lobehub#2716] [Development Question]
请问，接入 next-auth 用户系统之后，
1. 除了左上角换成用户的名称和头像之外，还有什么用吗？
2. 我能不能设定只有 sso 登陆的用户才可以使用呢？就替换掉 ACCESS_CODE 的功能？
3. 我能不能用一套通用的 setting. languageModel 配置来代替用户本地没有的配置？
4. 我能否统计某个用户使用的 token 数量？

翻了下文档，好像登入就是登入了，后继的功能完全没有做？
https://github.com/lobehub/lobe-chat/discussions/2022

我来确认下，如果真的没做的话，我就单纯在 nginx 上加上验证先凑着了，后期再看时间是否充裕再做后面的东西。

**Comment by arvinxx** [comment]
> 1. 除了左上角换成用户的名称和头像之外，还有什么用吗？

能实现 2 的拦截作用

> 2. 我能不能设定只有 sso 登陆的用户才可以使用呢？就替换掉 ACCESS_CODE 的功能？

你如果开起来 SSO 登录，默认就会拦截了。拦截的功能在 #1143 就做掉了的

> 3. 我能不能用一套通用的 setting. languageModel 配置来代替用户本地没有的配置？

目前 languageModel 可以用 ModelProvider 中的 XXXXXX_MODEL_LIST 来配置，详情查阅文档： https://lobehub.com/zh/docs/self-hosting/environment-variables/model-provider ，有什么缺失的可以提。

> 4. 我能否统计某个用户使用的 token 数量？

接入 SSO 以后，我们会默认将 SSO 的 userId 添加到 `/chat/competition` 的 user 字段上 （之前社区提的一个需求 #2659 ） ，所以只要你的 proxy 支持读到 user id自然就能统计

**Reply by DrayChou** [reply]
首先，感谢回复。

2. 再次验证了下2的拦截作用，更新代码，重启服务，清理浏览器缓存，然后左上角头像点开确实有登入按钮，然后点击主页上的开始聊天按钮，第一次没反应，多点了两下之后就进入聊天界面，然后就可以聊天了。重复验证了多次，可以复现，这跟昨天的体验一样，所以我才有这个提问。
3. 这个问题其实还是跟上面的2问题绑定的，这样直接写在 env 里也还能接受，我原来的设想是登入之后从我的服务器上下载对应的配置混合她本地的配置然后给他用。
4. 我看到代码里确实有这个了，我想想这个我怎么处理比较好。

谢谢解惑

**Reply by arvinxx** [reply]
> 点击主页上的开始聊天按钮，第一次没反应，多点了两下之后就进入聊天界面，然后就可以聊天了

你的意思是不开启 ACCESS_CODE，只开启 SSO 的模式，用户在没有登录的情况下可以直接用？如果是这样的话可以开个 issue ，然后给下环境变量的配置，我们来 check 下是否存在 bug。

**Reply by DrayChou** [reply]
> > 点击主页上的开始聊天按钮，第一次没反应，多点了两下之后就进入聊天界面，然后就可以聊天了
> 
> 你的意思是不开启 ACCESS_CODE，只开启 SSO 的模式，用户在没有登录的情况下可以直接用？如果是这样的话可以开个 issue ，然后给下环境变量的配置，我们来 check 下是否存在 bug。

不是，我的意思是在开启 SSO 模式下，非登陆用户应该是不可以用的，但是现在不是这样。
非登录下还是可以用的，而且用的还是我配置在 ENV 里的环境变量
